src/main/java/com/faforever/api/config/GlobalControllerExceptionHandler.java

```diff
@@ -4,44 +4,52 @@
 import com.faforever.api.config.error.ErrorResult;
 import com.faforever.api.error.ApiException;
 import com.faforever.api.error.ErrorCode;
-import com.google.common.collect.ImmutableMap;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
 import javax.validation.ValidationException;
-import java.io.Serializable;
 import java.text.MessageFormat;
 import java.util.Arrays;
-import java.util.Map;
 
 @ControllerAdvice
 // TODO implement a proper JSON-API error response
 class GlobalControllerExceptionHandler {
-  private Map<String, Serializable> errorResponse(String title, String message) {
-    ImmutableMap<String, Serializable> error = ImmutableMap.of(
-        "title", title,
-        "detail", message);
-    return ImmutableMap.of("errors", new ImmutableMap[]{error});
-  }
 
   @ExceptionHandler(ValidationException.class)
   @ResponseStatus(HttpStatus.UNPROCESSABLE_ENTITY)
   @ResponseBody
-  public Map<String, Serializable> processValidationException(ValidationException ex) {
-    return errorResponse(ErrorCode.VALIDATION_FAILED.getTitle(), ex.getMessage());
+  public ErrorResponse processValidationException(ValidationException ex) {
+    return new ErrorResponse().addError(new ErrorResult(
+      HttpStatus.UNPROCESSABLE_ENTITY.value(),
+      ErrorCode.VALIDATION_FAILED.getCode(),
+      ErrorCode.VALIDATION_FAILED.getTitle(),
+      ex.getMessage(),
+      null
+    ));
```

Comment:
Honestly I just refactored it. I didnt analyze whether it is possible.
This PR was pushed mainly to get some feedback if I am going right direction since jsonapi standard, spring security, oauth is something new for me